### PR TITLE
Completed Documentation with help of AI

### PR DESCRIPTION
**Pull request by porky11**
Documentation is complete now. So no breaking changes.

I still thought, I won't merge it already in case anybody wants to have a look.
Mostly because I also changed existing documentation, especially for consistency (it's not perfect yet).

But I'll probably just merge it myself tomorrow.

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -16,40 +16,53 @@ use crate::{ErrorKind, Renderer};
...
     Rgb8,
+    /// 32-bit RGBA image format (8 bits per channel, including alpha)
     Rgba8,
+    /// 8-bit grayscale image format
     Gray8,
 }
 
 bitflags! {
-    /// Image flags (eg. repeat, flip, mipmaps, etc.)
+    /// Represents a set of flags that modify the behavior of an image.
```
Comment (tronical) [inline src/image.rs:31]:
```suggestion
    /// Represents a set of flags that modify the behavior of an image when rendering.
```

File: src/image.rs [hunk 2]
```diff
@@ -16,40 +16,53 @@ use crate::{ErrorKind, Renderer};
...
     pub struct ImageFlags: u32 {
-        const GENERATE_MIPMAPS = 1;     // Generate mipmaps during creation of the image.
-        const REPEAT_X = 1 << 1;        // Repeat image in X direction.
-        const REPEAT_Y = 1 << 2;        // Repeat image in Y direction.
-        const FLIP_Y = 1 << 3;          // Flips (inverses) image in Y direction when rendered.
-        const PREMULTIPLIED = 1 << 4;   // Image data has premultiplied alpha.
-        const NEAREST = 1 << 5;         // Image interpolation is Nearest instead Linear
+        /// Generates mipmaps during the creation of the image.
+        const GENERATE_MIPMAPS = 1;
+        /// Repeats the image in the X direction when rendered.
```
Comment (tronical) [inline src/image.rs:36]:
```suggestion
        /// Repeats the image in the X direction when rendered into an area wider than the image.
```

(same for the other)

File: src/image.rs [hunk 3]
```diff
@@ -209,9 +224,7 @@ impl<T> ImageStore<T> {
         Ok(ImageId(self.0.insert((info, image))))
     }
 
-    ///
-    /// Reallocates the image without changing the id.
-    ///
+    // Reallocates the image without changing the id.
```
Comment (tronical) [inline src/image.rs:227]: Shouldn’t this be triple slash?

**Review by tronical — APPROVED** [review]

File: src/image.rs (hunk 3 above)
Comment (porky11) [reply src/image.rs:227]: The type `ImageStore` isn't exported, so it doesn't need to be a doccomment. Could also be removed completely.

**Pull request merged**